cmd/ttn-lw-cli/commands/end_device_templates.go

```diff
@@ -0,0 +1,451 @@
+		Use:   "from-device [flags]",
+		Short: "Create an end device template from an existing device",
+		Long: `Create an end device template from an existing device
+
+	This command takes end devices from stdin.`,
+		PersistentPreRunE: preRun(),
+		RunE: asBulk(func(cmd *cobra.Command, args []string) error {
+			if inputDecoder == nil {
+				return nil
+			}
+
+			forwardDeprecatedDeviceFlags(cmd.Flags())
+			paths := util.UpdateFieldMask(cmd.Flags(), selectEndDeviceFlags)
+
+			excludePaths := []string{
+				"ids.dev_addr",
+				"created_at",
+				"updated_at",
+				"network_server_address",
+				"application_server_address",
+				"join_server_address",
+			}
+			if appID, _ := cmd.Flags().GetBool("application-id"); !appID {
+				excludePaths = append(excludePaths, "ids.application_ids.application_id")
+			}
+			if devID, _ := cmd.Flags().GetBool("device-id"); !devID {
+				excludePaths = append(excludePaths, "ids.device_id")
+			}
+			if joinEUI, _ := cmd.Flags().GetBool("join-eui"); !joinEUI {
+				excludePaths = append(excludePaths, "ids.join_eui")
+			}
+			if devEUI, _ := cmd.Flags().GetBool("dev-eui"); !devEUI {
+				excludePaths = append(excludePaths, "ids.dev_eui")
+			}
+
+			var input ttnpb.EndDevice
+			decodedPaths, err := inputDecoder.Decode(&input)
+			if err != nil {
+				return err
+			}
+			decodedPaths = ttnpb.FlattenPaths(decodedPaths, endDeviceFlattenPaths)
+			decodedPaths = ttnpb.ExcludeFields(decodedPaths, excludePaths...)
+			paths = append(paths, decodedPaths...)
+
+			mappingKey, _ := cmd.Flags().GetString("mapping-key")
+			res := &ttnpb.EndDeviceTemplate{
+				FieldMask: pbtypes.FieldMask{
+					Paths: paths,
+				},
+				MappingKey: mappingKey,
+			}
+			res.EndDevice.SetFields(&input, paths...)
+
+			return io.Write(os.Stdout, config.OutputFormat, res)
+		}),
+	}
+	endDeviceTemplatesExecuteCommand = &cobra.Command{
+		Use:   "execute [flags]",
+		Short: "Execute the template to an end device",
+		Long: `Execute the template to an end device
+
+This command takes device templates from stdin.`,
+		PersistentPreRunE: preRun(),
+		RunE: asBulk(func(cmd *cobra.Command, args []string) error {
+			if inputDecoder == nil {
+				return nil
+			}
+
+			forwardDeprecatedDeviceFlags(cmd.Flags())
+
+			var input ttnpb.EndDeviceTemplate
+			_, err := inputDecoder.Decode(&input)
+			if err != nil {
+				return err
+			}
+
+			var device ttnpb.EndDevice
+			device.SetFields(&input.EndDevice, input.FieldMask.Paths...)
+			if err := util.SetFields(&device, setEndDeviceFlags); err != nil {
+				return err
+			}
+
+			return io.Write(os.Stdout, config.OutputFormat, &device)
+		}),
+	}
+	endDeviceTemplatesAssignEUIsCommand = &cobra.Command{
+		Use:     "assign-euis [join-eui] [start-dev-eui] [flags]",
+		Aliases: []string{"euis"},
+		Short:   "Assign JoinEUI and DevEUIs to end device templates",
+		Long: `Assign JoinEUI and DevEUIs to end device templates
+
+This command takes an end device template from stdin.`,
```

Comment:
Let's document that it either takes multiple templates, or generates `--count` for a single template (even though it technically allows both mutliple and `--count`)